WWZZSeparation/plotting/create_mjjmjjplots.cc

```diff
@@ -105,33 +104,31 @@ void create_mjjmjjplots(
 	h2_ZZ->Scale(1./ h2_ZZ->Integral());
 
 	// WW = blue, ZZ = red
-	h1_WW->SetLineColor(4);	
-	h1_ZZ->SetLineColor(2);	
-	h2_WW->SetLineColor(4);	
-	h2_ZZ->SetLineColor(2);	
+	h1_WW->SetLineColor(4);
+	h1_ZZ->SetLineColor(2);
+	h2_WW->SetLineColor(4);
+	h2_ZZ->SetLineColor(2);
 
 
 	// Do drawing and saving of plots
 	TCanvas *canvas_h1 = new TCanvas("can_h1", "", 0, 0, 600, 600);
 	h1_WW->Draw("hist");
 	h1_ZZ->Draw("hist same");
 
-	string plot_name_h1 = input_directory + "/m.pdf";
+	string plot_name_h1 = input_dir_base + "/m.pdf";
 	canvas_h1->Print(plot_name_h1.c_str());
 
 	TCanvas *canvas_h2 = new TCanvas("can_h2", "", 0, 0, 800, 800);
 	h2_WW->Draw("box");
 	h2_ZZ->Draw("box same");
 
-	string plot_name_h2 = input_directory + "/m_m.pdf";
+	string plot_name_h2 = input_dir_base + "/m_m.pdf";
 	canvas_h2->Print(plot_name_h2.c_str());
 
 	// Avoid memory leaks
 	delete h1_WW;
 	delete h2_WW;
 	delete h1_ZZ;
 	delete h2_ZZ;
-	delete canvas_h1;
```

Comment:
You had a double delete problem before. But now you have a double memory leak here.